### PR TITLE
Schema fee center

### PR DESCRIPTION
**Pull request by DedrickEnc**
This PR adds an initial schema and API to continue the feature developing
Here are things to notice in this PR : 

- The table ```cost_center``` and ```profit_center``` are deleted
-  A new table ```fee_center``` is created```
- The modules cost center and profit center are deleted
- The service module is updated to fit with the  changes
- The delete functionality is not yet implemented and the integration  test for this functionality is ignored, because we did not talk about that during the design. 
Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by DedrickEnc** [comment]
@sfount @jniles

File: server/controllers/finance/feeCenter.js
Comment (jniles) [inline server/controllers/finance/feeCenter.js]: Just a note, since these values never change, `const` is more appropriate.  ESLint should alert you of this - if it isn't, be sure to configure your text editor to use it.

File: server/controllers/finance/feeCenter.js
Comment (jniles) [inline server/controllers/finance/feeCenter.js]: We make the entire server globally `strict`.  No need to have this `use strict` directive.

See https://github.com/IMA-WorldHealth/bhima-2.X/pull/1105#discussion_r98899888 as well.

File: server/controllers/finance/feeCenter.js
Comment (jniles) [inline server/controllers/finance/feeCenter.js]: This statement would probably be more efficient if you used `SELECT DISTINCT`.  That way, MySQL could eliminate duplicates before searching through the array.

File: server/controllers/finance/feeCenter.js
Comment (jniles) [inline server/controllers/finance/feeCenter.js]: Giving example code in your comments is an excellent idea.  It makes the usage clear, and it can potentially be extracted and run by a documentation program to make sure that the function behaves as expected.

However, this code will not work, because the file name is `finance/feeCenter.js`.  Could you either remove the misleading comment or update the name?   This problem has been raised in another review https://github.com/IMA-WorldHealth/bhima-2.X/pull/1105#discussion_r98898911.

File: server/controllers/finance/feeCenter.js
Comment (jniles) [inline server/controllers/finance/feeCenter.js]: Since this variable binding never changes, it would be more appropriate to use `const`.

File: server/controllers/finance/feeCenter.js
Comment (jniles) [inline server/controllers/finance/feeCenter.js]: This is a pretty complicated query to understand.  It would be more readable if you did the `SUM()` and `IF()` functions on different statements.

Alternatively, you could just do a multiplication by `-1` to inverse the amounts, with a comment explaining why it should be inverted.
```js
const sql = `
  SELECT id, IF(is_cost = 1, value, value * -1) AS value
  FROM (
    SELECT SUM(gl.debit_equiv - gl.credit_equiv) AS value, f.is_cost, f.id
    FROM general_ledger AS gl JOIN fee_center AS f
      ON gl.fc_id = f.id
    WHERE gl.fc_id = ?
    GROUP BY gl.fc_id
  )z;
`
```
This query may also be more performant by only using a single `SUM()`.

File: server/lib/filter.js
Comment (jniles) [inline server/lib/filter.js:21]: :+1:

File: server/lib/filter.js
Comment (jniles) [inline server/lib/filter.js:46]: :+1:

File: server/models/schema.sql
Comment (jniles) [inline server/models/schema.sql:347]: Will this always contain costs?  Or can we assign profits as well?  If so, we should find a better table name before we get too far.

File: test/integration/feeCenter.js
Comment (jniles) [inline test/integration/feeCenter.js]: This comment is no longer needed.

File: test/integration/feeCenter.js
Comment (jniles) [inline test/integration/feeCenter.js:171]: Why are these skipped?  Does the delete route not work?

**Review by jniles — COMMENTED** [review]
This PR has gone through a few iterations and this is the third or fourth review.  There are still a few issues that have carried over from previous reviews that should be addressed, either in code or in comments.

I've made a few suggestions for improvements to your code, but nothing is blocking.  Let me know what you think of the suggestions and we can merge or update the code.

@sfount has been following the design of Fee Centers more closely.   I really only looked at the code - he should take a look and make sure it fits with the design vision.

File: server/models/schema.sql
Comment (DedrickEnc) [reply server/models/schema.sql:347]: There is only auxiliary cost center and not auxiliary profit center, I will propose an issue to remove this concept of cost and profit. so we will keep only fee center that will help to restructure this table also.

Comment (jniles) [reply server/models/schema.sql:347]: Sounds good.  We'll get this in before that change occurs.

File: test/integration/feeCenter.js
Comment (DedrickEnc) [reply test/integration/feeCenter.js:171]: We did not plan a deletion operation for fee center so they are there just in case there is a different decision. So there is no route at the server to delete a fee center for now.

**Comment by DedrickEnc** [comment]
@jniles

**Review by jniles — APPROVED** [review]
Changes LGTM.

**Pull request merged**
